### PR TITLE
[Feature] Grading System - Student Exam Summary

### PR DESCRIPTION
**Pull request by ivohashamov**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, editor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/server.html).
- [x] Server: I added multiple integration tests (Spring) related to the features (with a high test coverage)
- [x] Server: I implemented the changes with a good performance and prevented too many database calls
- [x] Server: I documented the Java code using JavaDoc style.
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I added multiple integration tests (Jest) related to the features (with a high test coverage)
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I added multiple screenshots/screencasts of my UI changes
- [x] Client: I translated all the newly inserted strings into German and English

### Motivation and Context

This PR integrates the grading system in the student exam summary page. Students are now able to see the grade they've achieved on the exam (respectively the bonus points they've gained if the exam grades are of type bonus). The bonus points information is also now hidden, as long as there are no exercises that give bonus points (which is almost always the case). 

### Description

I've added some conditions to already present columns in the exam points summary component.
Also, I've added an additional section that displays the grade/bonus points of the student.
In a follow-up, a link will be created to the grade key, so a student can see the scale and where he/she fits into it.

### Steps for Testing

1. Log in to Artemis as an instructor
2. Set a grade key of type 'Grade' for an exam with student submissions in it.
3. Log in as one of the students with submissions in the said exam and go to the exam summary page (Course Overview -> Course -> Exams -> Exam name)
a) check if the corresponding grade is present (red if the student has failed the exam and green if he/she passed)
b) if none of the exam's exercises have bonus points, no column for bonus points should be displayed in the summary
c) test exporting (Colors don't get maintained unless the 'Background graphics' option is ticked in the export GUI).
d) check if any errors are getting displayed in the console/network tab
4. Now switch the grade ket to type 'Bonus' (again as an instructor)
5. Go to the same exam summary page and check the same steps in 3 a) - d), where instead of grade outcome a bonus points outcome should be displayed (without any colors).
6. Finally delete the grade key and go to the same overview page to test if any errors pop up if no grade key is present.

### Test Coverage

GradeStepResource.java: 100%
exam-points-summary.component.ts: 94.52%

### Screenshots

Screenshot 2021-05-24 175701
Screenshot 2021-05-24 231345
Screenshot 2021-05-24 175614

### CODE REVIEW

**Review by imstreetspirit — DISMISSED** [review]
tested on ts3, seems working as described. code looks good to me.

**Review by KiraNickel — APPROVED** [review]
Tested on ts3, everything works as described 👍🏼

File: src/main/webapp/i18n/de/exam.json [hunk 1]
```diff
@@ -21,8 +21,12 @@
                     "yourPoints": "Deine Punkte",
                     "maxPoints": "Punkte",
                     "bonus": "Bonuspunkte",
-                    "youAchieved": "Du hast {{achievedPoints}} Punkte von möglichen {{normalPoints}} normalen Punkten und {{bonusPoints}} Bonuspunkten erreicht"
-                }
+                    "youAchievedWithBonus": "Du hast {{achievedPoints}} Punkte von möglichen {{normalPoints}} normalen Punkten und {{bonusPoints}} Bonuspunkten erreicht",
```
Comment (SimonEntholzer) [inline src/main/webapp/i18n/de/exam.json:24]: I would reformulate this: 
```suggestion
                    "youAchievedWithBonus": "Du hast {{achievedPoints}} Punkte von {{normalPoints}} möglichen Punkten, und {{bonusPoints}} Bonuspunkten erreicht.",
```

File: src/main/webapp/i18n/de/exam.json [hunk 2]
```diff
@@ -21,8 +21,12 @@
...
+                    "youAchieved": "Du hast {{achievedPoints}} Punkte von möglichen {{normalPoints}} normalen Punkten",
```
Comment (SimonEntholzer) [inline src/main/webapp/i18n/de/exam.json:25]: I would leave out the 'normal'
```suggestion
                    "youAchieved": "Du hast {{achievedPoints}} von möglichen {{normalPoints}} Punkten",
```

**Review by SimonEntholzer — CHANGES_REQUESTED** [review]
Tested on TS3 and everything worked as intended, The code looks good as well. Just found two small things:

**Review by KiraNickel — DISMISSED** [review]
Re-approve, tested again on ts3 :)

**Review by SimonEntholzer — APPROVED** [review]

**Review by argertboja — COMMENTED** [review]
Tested on TS3, everything works as expected 👍🏼

**Pull request merged**